src/main/java/persistence/entity/DefaultEntityManager.java

```diff
@@ -0,0 +1,65 @@
+package persistence.entity;
+
+
+import java.sql.ResultSet;
+import java.util.List;
+import persistence.exception.NotFoundException;
+import persistence.mapper.EntityRowsMapper;
+import persistence.mapper.EntitySingleMapper;
+import persistence.jdbc.JdbcTemplate;
+import persistence.meta.EntityMeta;
+import persistence.sql.QueryGenerator;
+
+
+public class DefaultEntityManager implements EntityManager {
+    private final JdbcTemplate jdbcTemplate;
+
+    public DefaultEntityManager(JdbcTemplate jdbcTemplate) {
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
+    @Override
+    public <T> T find(Class<T> clazz, Long id) {
+        if (id == null) {
+            throw new NotFoundException("id가 널이면 안 됩니다.");
+        }
+
+        final EntityMeta entityMeta = new EntityMeta(clazz);
```

Comment:
```suggestion
    @Override
    public <T> T find(EntityMeta<T> entityMeta, Long id) {
        if (id == null) {
            throw new NotFoundException("id가 널이면 안 됩니다.");
        }
```

find 할 때마다 EntityMeta 를 매번 생성하는게 아니라,
Repository 의 생성자에서 clazz 를 넘겨받고 EntityMeta 를 생성해서,
멤버로 가지고 있으면 어떨까 제안해봅니다.

이런 구조를 생각하면, JPA 를 쓸 때 왜 Entity 클래스 별로,
별도의 Repository 를 선언해줘야하는 지 깊게 이해할 수 있게 되지 않을까 생각합니다.
반면 EntityManager 는 domain 에 선언한 Entity 가 여러개라 하더라도,
같은 EntityManager 를 쓸 수가 있지요.

Reply:
확실하게 멤버로 가지고 있으니까 같은 EntityManager를 쓰면서 메모리 낭비가 덜 하겠군요!